File: 100-days-of-code/Beginner/Day02/bmi.py
def main():
    print("Welcome to Day 2!")
    print("First, let's calculate your BMI!")
    while True:
        height = input("What's your height in meters?\n")
        weight = input("What's your weight in kilograms?\n")

        try:
            height = float(height)
            weight = float(weight)
            break
        except (TypeError, ValueError) as e:
            print("Please enter a number for both height and weight")
            print(e)

    bmi = bmi_calc(weight, height)
    print(f"Your BMI is {bmi:.2f}")


def bmi_calc(w: float, h: float):
    return w / (h * h)


if __name__ == "__main__":
    main()

File: 100-days-of-code/Beginner/Day02/src/bmi.rs
use std::io;

pub fn bmi_prompt() -> io::Result<f64> {
    println!("Welcome to Day 2!");
    println!("First, let's calculate your BMI!");

    let mut input = String::new();
    let height = loop {
        input.clear();
        println!("What's your height in metres?");
        io::stdin().read_line(&mut input)?;
        match input.trim().parse::<f64>() {
            Ok(num) => break num,
            Err(err) => eprintln!("{}", err),
        }
    };

    let width = loop {
        input.clear();
        println!("What's your weight in kilograms?");
        io::stdin().read_line(&mut input)?;
        match input.trim().parse::<f64>() {
            Ok(num) => break num,
            Err(err) => eprintln!("{}", err),
        }
    };

    let bmi = bmi_calc(width, height);

    Ok(bmi)
}

fn bmi_calc(w: f64, h: f64) -> f64 {
    w / (h * h)
}
